packages/atomic/src/components/commerce/atomic-commerce-interface/atomic-commerce-interface.tsx

```diff
@@ -0,0 +1,480 @@
+   * ```
+   */
+  @ArrayProp()
+  @Prop({mutable: true})
+  public fieldsToInclude: string[] | string = '[]';
+
+  /**
+   * Whether analytics should be enabled.
+   */
+  @Prop({reflect: true}) public analytics = true;
+
+  /**
+   * The severity level of the messages to log in the console.
+   */
+  @Prop({reflect: true}) public logLevel?: LogLevel;
+
+  /**
+   * The compatibility JSON version for i18next to use (see [i18next Migration Guide](https://www.i18next.com/misc/migration-guide#v20.x.x-to-v21.0.0)).
+   */
+  @Prop() public localizationCompatibilityVersion: i18nCompatibilityVersion =
+    'v3';
+
+  /**
+   * the commerce interface i18next instance.
+   */
+  @Prop() public i18n: i18n = i18next.createInstance();
+
+  /**
+   * the commerce interface language.
+   */
+  @Prop({reflect: true}) public language = 'en';
+
+  /**
+   * The commerce interface headless engine.
+   */
+  @Prop({mutable: true}) public engine?: CommerceEngine;
+
+  /**
+   * Whether the state should be reflected in the URL parameters.
+   */
+  @Prop({reflect: true}) public reflectStateInUrl = true;
+
+  /**
+   * The CSS selector for the container where the interface will scroll back to.
+   */
+  @Prop({reflect: true}) public scrollContainer = 'atomic-commerce-interface';
+
+  /**
+   * The language assets path. By default, this will be a relative URL pointing to `./lang`.
+   *
+   * Example: "/mypublicpath/languages"
+   *
+   */
+  @Prop({reflect: true}) public languageAssetsPath = './lang';
+
+  /**
+   * The icon assets path. By default, this will be a relative URL pointing to `./assets`.
+   *
+   * Example: "/mypublicpath/icons"
+   *
+   */
+  @Prop({reflect: true}) public iconAssetsPath = './assets';
+
+  /**
+   * The value to set the [nonce](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce) attribute to on inline script and style elements generated by this interface and its child components.
+   * If your application is served with a Content Security Policy (CSP) that doesn't include the `script-src: 'unsafe-inline'` or `style-src: 'unsafe-inline'` directives,
+   * you should ensure that your application server generates a new nonce on every page load and uses the generated value to set this prop and serve the corresponding CSP response headers
+   * (i.e., script-src 'nonce-<YOUR_GENERATED_NONCE>' and style-src 'nonce-<YOUR_GENERATED_NONCE>').
+   * Otherwise you may see console errors such as
+   *  - Refused to execute inline script because it violates the following Content Security Policy directive: [...]
+   *  - Refused to apply inline style because it violates the following Content Security Policy directive: [...].
+   * @example:
+   * ```html
+   * <script nonce="<YOUR_GENERATED_NONCE>">
+   *  import {setNonce} from '@coveo/atomic';
+   *  setNonce('<YOUR_GENERATED_NONCE>');
+   * </script>
+   * ```
+   */
+  @Prop({reflect: true}) public CspNonce?: string;
```

Comment:
`cspNonce` ?

Reply:
Thats the nonce support added by @louis-bompart here in #3791
I included it since I assumed that we want to support this here too

Reply:
good call.